lib/cache/BaseCache.ts

```diff
@@ -71,7 +74,7 @@ export class BaseCache<T extends CacheEntry> {
   }
 
   protected ensureCacheDirectory(): void {
-    if (!fs.existsSync(this.cacheDir)) {
+    if (this.enableCaching && !fs.existsSync(this.cacheDir)) {
```

Comment:
This is an antipattern. Since this method is `protected`, it can only be called from other cache objects. If we just don't init a cache, the directory won't be created